src/python/pants/binaries/binary_util.py

```diff
@@ -237,29 +238,29 @@ def _fetch_binary(self, name, binary_path):
     bootstrap_dir = os.path.realpath(os.path.expanduser(self._pants_bootstrapdir))
     bootstrapped_binary_path = os.path.join(bootstrap_dir, binary_path)
     if not os.path.exists(bootstrapped_binary_path):
-      downloadpath = bootstrapped_binary_path + '~'
-      try:
+      safe_mkdir_for(bootstrapped_binary_path)
+      with temporary_dir(root_dir=os.path.dirname(bootstrapped_binary_path)) as tmpdir:
+        downloadpath = os.path.join(tmpdir, 'download')
         with self._select_binary_stream(name, binary_path) as stream:
           with safe_open(downloadpath, 'wb') as bootstrapped_binary:
             bootstrapped_binary.write(stream())
           os.rename(downloadpath, bootstrapped_binary_path)
           chmod_plus_x(bootstrapped_binary_path)
-      finally:
-        safe_delete(downloadpath)
 
     logger.debug('Selected {binary} binary bootstrapped to: {path}'
                  .format(binary=name, path=bootstrapped_binary_path))
     return bootstrapped_binary_path
 
-  def _compare_file_checksums(self, filepath, checksum=None, digest=None):
+  @staticmethod
+  def _compare_file_checksums(filepath, checksum=None, digest=None):
     digest = digest or hashlib.sha1()
 
     if os.path.isfile(filepath) and checksum:
       return hash_file(filepath, digest=digest) == checksum
 
     return os.path.isfile(filepath)
 
-  def is_bin_valid(self, basepath, binary_file_specs=[]):
+  def is_bin_valid(self, basepath, binary_file_specs=tuple()):
```

Comment:
If using `tuple()` instead of `()` was deliberate, no beef, but it was slightly surprising to read.

Reply:
I think I forgot that `()` is a valid literal. Changed.